### ISSUE TITLE
Cannot validate numericality with dynamic values

### ISSUE THREAD

**Issue by mochetts** [thoughtbot/shoulda-matchers#1583]
This doesn't work in tests (even though the validation works fine):

## Model
```rb
class Product < ApplicationRecord
  validates :cost, numericality: {greater_than_or_equal_to: 0}
  validates :value, numericality: {greater_than_or_equal_to: :cost}
end
```

## Test
```rb
it { should validate_numericality_of(:value).is_greater_than_or_equal_to(:cost) }
```

## Error

```bash
2) Product validations is expected to validate that :value looks like a number greater than or equal to cost
     Failure/Error: it { should validate_numericality_of(:value).is_greater_than_or_equal_to(:cost) }
     
     NoMethodError:
       undefined method `+' for :cost:Symbol
     # /Users/martin/.gem/ruby/3.2.2/gems/shoulda-matchers-4.5.1/lib/shoulda/matchers/active_model/numericality_matchers/comparison_matcher.rb:141:in `block in diffs_to_compare'
```

**Comment by matsales28** [comment]
Hey @mochetts, thanks for opening this issue. I'll take a look as soon as I have some time!

**Comment by matsales28** [comment]
@mochetts This was fixed on #1552. We'll have a new release soon that introduces those changes and also fix this problem.

I'm closing this issue. Feel free to re-open it if you think there's something we can do.
